state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -942,15 +948,42 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			asset,
 			deposit_address: deposit_address.clone(),
 			deposit_fetch_id: None,
-			amount,
+			amount: deposit_amount,
 		});
 
 		let channel_id = deposit_channel_details.deposit_channel.channel_id;
 		Self::deposit_event(Event::<T, I>::DepositFetchesScheduled { channel_id, asset });
 
+		// Shave off the fetch fee from the deposit amount.
+		// For now, we take ingress and egress fees here. This is a simplification: we should
+		// really take the egress fee at the point of egress.
+		let tracked_data = T::ChainTracking::get_tracked_data();
+		let tx_fees = tracked_data
+			.estimate_ingress_fee(deposit_channel_details.deposit_channel.asset) +
+			tracked_data.estimate_egress_fee(deposit_channel_details.deposit_channel.asset);
```

Comment:
The problem is that the error in this impl is huge because if we swap from cheap chain (e.g. polkadot) to expensive chain (like bitcoin), we basically deduct no fees for egress. 

I think the implementation would be still relatively not complex if we just move this calculation to the match statement below.

Reply:
As discussed I moved the egress fee shaving to when we do the egress.